crates/sui-framework/packages/bridge/sources/limiter.move

```diff
@@ -0,0 +1,192 @@
+        record.total_amount = record.total_amount + notional_amount;
+    }
+
+    #[test]
+    fun test_24_hours_windows() {
+        let limiter = TransferLimiter {
+            transfer_limits: vec_map::empty(),
+            notional_values: vec_map::empty(),
+            // Per hour transfer amount for each bridge route
+            transfer_records: vec_map::empty(),
+        };
+
+        let route = chain_ids::get_route(chain_ids::sui_devnet(), chain_ids::eth_sepolia());
+
+        // Global transfer limit is 1M USD
+        vec_map::insert(&mut limiter.transfer_limits, route, 1_000_000);
+        // Notional price for ETH is 10 USD
+        vec_map::insert(&mut limiter.notional_values, type_name::get<ETH>(), 10);
+
+        let scenario = test_scenario::begin(@0x1);
+        let ctx = test_scenario::ctx(&mut scenario);
+        let clock = clock::create_for_testing(ctx);
+        clock::set_for_testing(&mut clock, 1706288001377);
+
+        check_and_record_transfer<ETH>(&clock, &mut limiter, route, 1_000_000_000_000);
+
+        let record = vec_map::get(&limiter.transfer_records, &route);
+        assert!(record.total_amount == 10000 * 10, 0);
+
+        // transfer 1000 ETH every hour, the 24 hours totol should be 24000 * 10
+        let i = 0;
+        while (i < 50) {
+            clock::increment_for_testing(&mut clock, 60 * 60 * 1000);
+            check_and_record_transfer<ETH>(&clock, &mut limiter, route, 100_000_000_000);
+            i = i + 1;
+        };
+
+        let record = vec_map::get(&limiter.transfer_records, &route);
+        assert!(record.total_amount == 24000 * 10, 0);
+
+        destroy(limiter);
+
+        clock::destroy_for_testing(clock);
+        test_scenario::end(scenario);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = ETransferAmountExceedLimit)]
+    fun test_exceed_limit() {
+        let limiter = TransferLimiter {
+            transfer_limits: vec_map::empty(),
+            notional_values: vec_map::empty(),
+            // Per hour transfer amount for each bridge route
+            transfer_records: vec_map::empty(),
+        };
+
+        let route = chain_ids::get_route(chain_ids::sui_devnet(), chain_ids::eth_sepolia());
+        // Global transfer limit is 1M USD
+        vec_map::insert(&mut limiter.transfer_limits, route, 1_000_000);
+        // Notional price for ETH is 10 USD
+        vec_map::insert(&mut limiter.notional_values, type_name::get<ETH>(), 10);
+
+        let scenario = test_scenario::begin(@0x1);
+        let ctx = test_scenario::ctx(&mut scenario);
+        let clock = clock::create_for_testing(ctx);
+        clock::set_for_testing(&mut clock, 1706288001377);
+
+        check_and_record_transfer<ETH>(&clock, &mut limiter, route, 9_000_000_000_000);
+
+        let record = vec_map::get(&limiter.transfer_records, &route);
+        assert!(record.total_amount == 90000 * 10, 0);
+
+        // tx should fail after 10 iteration when total amount exceed 1000000 * 10^8
```

Comment:
with this while loop we don't know if it actually aborts in the 10th iteration

Reply:
sadly it's hard to catch error in move, will think of a better way to test this

Reply:
yeah .. I think we can use two tests here with the same setup. But one with 9 loops and Ok, the other one with 10 loops aborts